diff -urN gpsd-3.25.orig/gpsd/libgpsd_core.c gpsd-3.25/gpsd/libgpsd_core.c
--- gpsd-3.25.orig/gpsd/libgpsd_core.c	2023-01-10 23:38:26.000000000 +0100
+++ gpsd-3.25/gpsd/libgpsd_core.c	2026-04-22 14:06:09.498017456 +0200
@@ -180,17 +180,8 @@
     // this was carzy expensive, just fix the bad log calls
     // gps_visibilize(outbuf, outlen, buf, strlen(buf));
 
-    if (getpid() == getsid(getpid())) {
-        // I think this calls syslog() only when daemonized
-        syslog(level, "%s",  outbuf);
-    } else if (NULL != errout->report) {
-        // we are a thread, use report()?
-        // FIXME: is POSIX syslog() thread safe?
-        errout->report(outbuf);
-    } else {
-        // foreground, use stderr?
-        (void)fputs(outbuf, stderr);
-    }
+    // always log to syslog, even with -N
+    syslog(level, "%s",  outbuf);
     gpsd_release_reporting_lock();
 #endif  // !SQUELCH_ENABLE
 }
