diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index 2406658da..4226a98ca 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -1508,7 +1496,7 @@ static int hostapd_setup_bss(struct hostapd_data *hapd, int first,
 				   conf->iface, addr, hapd,
 				   &hapd->drv_priv, force_ifname, if_addr,
 				   conf->bridge[0] ? conf->bridge : NULL,
-				   first == -1)) {
+				   1)) {
 			wpa_printf(MSG_ERROR, "Failed to add BSS (BSSID="
 				   MACSTR ")", MAC2STR(hapd->own_addr));
 			hapd->interface_added = 0;
