From 8e230d3f2e1e33691998862e99ab4e13410527b9 Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Wed, 8 Jan 2025 07:49:06 +0100
Subject: [PATCH 01/16] Drop redundant initial call to localtime_r()
Organization: Wires

Fixes #94

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/syslogd.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/syslogd.c b/src/syslogd.c
index 17115fc..5ad4dbc 100644
--- a/src/syslogd.c
+++ b/src/syslogd.c
@@ -1508,7 +1508,6 @@ void printsys(char *msg)
 			 */
 			if (KeepKernTime || !sys_seqno_init) {
 				now = boot_time + ustime / 1000000;
-				localtime_r(&now, &buffer.timestamp.tm);
 			} else {
 				struct timeval tv;
 
-- 
2.43.0

