From c88e28eb099017e43f3f24d238d3e350bc925b4b Mon Sep 17 00:00:00 2001
From: Michal Vasko <mvasko@cesnet.cz>
Date: Thu, 16 Jan 2025 12:15:00 +0100
Subject: [PATCH 07/27] doc UPDATE rewording of the new node of lyd_new_path
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Wires

Signed-off-by: Mattias Walström <lazzer@gmail.com>
---
 src/tree_data.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tree_data.h b/src/tree_data.h
index 052ee0963..782c29341 100644
--- a/src/tree_data.h
+++ b/src/tree_data.h
@@ -1591,7 +1591,7 @@ LIBYANG_API_DECL LY_ERR lyd_new_path(struct lyd_node *parent, const struct ly_ct
  * @param[in] value_type Anyxml/anydata node @p value type.
  * @param[in] options Bitmask of options, see @ref newvaloptions.
  * @param[out] new_parent Optional first parent node created. If only one node was created, equals to @p new_node.
- * @param[out] new_node Optional last node created (in case of a list, it is the list instance, not any of the key nodes).
+ * @param[out] new_node Optional target node of @p path (the last created node, the list instance in case of a list).
  * @return LY_SUCCESS on success.
  * @return LY_EEXIST if the final node to create exists (unless ::LYD_NEW_PATH_UPDATE is used).
  * @return LY_EINVAL on invalid arguments including invalid @p path.
-- 
2.43.0

