From 171a0e8ee8f636d3b464c31882da4fbeb5977581 Mon Sep 17 00:00:00 2001
From: Michal Vasko <mvasko@cesnet.cz>
Date: Thu, 16 Jan 2025 14:43:08 +0100
Subject: [PATCH 08/27] doc UPDATE clarify state list creation for lyd_new_path
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Wires

Signed-off-by: Mattias Walström <lazzer@gmail.com>
---
 src/tree_data.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tree_data.h b/src/tree_data.h
index 782c29341..fc0881aeb 100644
--- a/src/tree_data.h
+++ b/src/tree_data.h
@@ -1554,8 +1554,8 @@ LIBYANG_API_DECL LY_ERR lyd_new_attr2(struct lyd_node *parent, const char *modul
  * and @p value is set, the predicate is preferred.
  *
  * For key-less lists, positional predicates must be used (indices starting from 1). For non-configuration leaf-lists
- * either positional predicate can be used or leaf-list predicate, when an instance is always created at the end.
- * If no predicate is used for these nodes, they are always created.
+ * (or lists) either positional predicate can be used or leaf-list (or key) predicate, when an instance is always
+ * created at the end. If no predicate is used for these nodes, they are always created.
  *
  * @param[in] parent Data parent to add to/modify, can be NULL. Note that in case a first top-level sibling is used,
  * it may no longer be first if @p path is absolute and starts with a non-existing top-level node inserted
-- 
2.43.0

