From 3b54bdc4546d7009cc0443a8c5889b1075253f7b Mon Sep 17 00:00:00 2001
From: Michal Vasko <mvasko@cesnet.cz>
Date: Thu, 30 Jan 2025 16:05:48 +0100
Subject: [PATCH 22/27] xpath UPDATE local node preference to extension nodes
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Wires

Signed-off-by: Mattias Walström <lazzer@gmail.com>
---
 src/xpath.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/xpath.c b/src/xpath.c
index 1e7130a8c..da4ce09db 100644
--- a/src/xpath.c
+++ b/src/xpath.c
@@ -6802,7 +6802,8 @@ moveto_scnode(struct lyxp_set *set, const struct lys_module *moveto_mod, const c
             }
         }
 
-        if (moveto_mod && ncname && ((axis == LYXP_AXIS_DESCENDANT) || (axis == LYXP_AXIS_CHILD)) &&
+        /* only consider extension nodes after no local ones were found */
+        if ((orig_used == set->used) && moveto_mod && ncname && ((axis == LYXP_AXIS_DESCENDANT) || (axis == LYXP_AXIS_CHILD)) &&
                 (set->val.scnodes[i].type == LYXP_NODE_ELEM) && !ly_nested_ext_schema(NULL, set->val.scnodes[i].scnode,
                 moveto_mod->name, strlen(moveto_mod->name), LY_VALUE_JSON, NULL, ncname, strlen(ncname), &iter, NULL)) {
             /* there is a matching node from an extension, use it */
-- 
2.43.0

