From b66f95bc9d45c306c7e5f93df457c1ff326a5f1f Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Tue, 7 Feb 2023 18:29:44 +0100
Subject: [PATCH 1/4] util: clixon_util_regexp requires libxml2
Organization: Addiva Elektronik

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 util/Makefile.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/util/Makefile.in b/util/Makefile.in
index 552cedda..fbb435af 100644
--- a/util/Makefile.in
+++ b/util/Makefile.in
@@ -95,7 +95,9 @@ APPSRC   += clixon_util_yang.c
 APPSRC   += clixon_util_xpath.c
 APPSRC   += clixon_util_path.c
 APPSRC   += clixon_util_datastore.c
+ifdef with_libxml2
 APPSRC   += clixon_util_regexp.c
+endif
 APPSRC   += clixon_util_socket.c
 APPSRC   += clixon_util_validate.c
 APPSRC   += clixon_util_dispatcher.c 
-- 
2.34.1

