From 8181ce6caf5ee49e30a02db9695647bd7bc73838 Mon Sep 17 00:00:00 2001
From: Michal Vasko <mvasko@cesnet.cz>
Date: Tue, 2 Jul 2024 12:11:30 +0200
Subject: [PATCH 5/9] modinfo BUGFIX memory leak
Organization: Addiva Elektronik

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/modinfo.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/modinfo.c b/src/modinfo.c
index 949c321b..9edb4498 100644
--- a/src/modinfo.c
+++ b/src/modinfo.c
@@ -1042,6 +1042,10 @@ sr_xpath_oper_data_get(struct sr_mod_info_mod_s *mod, const char *xpath, const c
 cleanup:
     lyd_free_tree(parent_dup);
     free(parent_path);
+    if (err_info) {
+        lyd_free_all(*oper_data);
+        *oper_data = NULL;
+    }
     return err_info;
 }
 
-- 
2.43.0

