From d61b847e557e5162163dffd1abd4b5d9bedc6d7a Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Sun, 2 Apr 2023 22:17:09 +0200
Subject: [PATCH 2/4] rsync: fix check for missing dst when dst is a directory
Organization: Addiva Elektronik

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/rsync.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/rsync.c b/src/rsync.c
index 92b8d63..695fac3 100644
--- a/src/rsync.c
+++ b/src/rsync.c
@@ -77,7 +77,9 @@ int rsync(char *src, char *dst, int opt, int (*filter)(const char *file))
 	char **files;		/* Array of file names. */
 	struct stat st;
 
-	if (!fisdir(dst))
+	errno = 0;
+
+	if (stat(dst, &st) && fisslashdir(dst))
 		makedir(dst, 0755);
 
 	if (!fisdir(src)) {
-- 
2.34.1

