diff --git a/lib/northbound.c b/lib/northbound.c
index 6ff5c24bd1..c6cca523b2 100644
--- a/lib/northbound.c
+++ b/lib/northbound.c
@@ -581,7 +581,7 @@ void nb_config_diff(const struct nb_config *config1,
 		char *s;
 
 		if (!lyd_print_mem(&s, diff, LYD_JSON,
-				   LYD_PRINT_WITHSIBLINGS | LYD_PRINT_WD_ALL)) {
+				   LYD_PRINT_SIBLINGS | LYD_PRINT_WD_ALL)) {
 			zlog_debug("%s: %s", __func__, s);
 			free(s);
 		}
@@ -2283,7 +2283,7 @@ bool nb_operation_is_valid(enum nb_operation operation,
 			if (sleaf->when)
 				return true;
 			if (CHECK_FLAG(sleaf->flags, LYS_MAND_TRUE)
-			    || sleaf->dflt)
+			    || sleaf->dflt.str)
 				return false;
 			break;
 		case LYS_CONTAINER:
diff --git a/lib/northbound_cli.c b/lib/northbound_cli.c
index 8003679ed5..0997b5f8c5 100644
--- a/lib/northbound_cli.c
+++ b/lib/northbound_cli.c
@@ -634,7 +634,7 @@ static int nb_cli_show_config_libyang(struct vty *vty, LYD_FORMAT format,
 		return CMD_WARNING;
 	}
 
-	SET_FLAG(options, LYD_PRINT_WITHSIBLINGS);
+	SET_FLAG(options, LYD_PRINT_SIBLINGS);
 	if (with_defaults)
 		SET_FLAG(options, LYD_PRINT_WD_ALL);
 	else
@@ -1443,7 +1443,7 @@ DEFPY (show_yang_operational_data,
 	struct ly_ctx *ly_ctx;
 	struct lyd_node *dnode;
 	char *strp;
-	uint32_t print_options = LYD_PRINT_WITHSIBLINGS;
+	uint32_t print_options = LYD_PRINT_SIBLINGS;
 	int ret;
 
 	if (xml)
diff --git a/lib/northbound_db.c b/lib/northbound_db.c
index 74abcde955..7d51f39291 100644
--- a/lib/northbound_db.c
+++ b/lib/northbound_db.c
@@ -79,7 +79,7 @@ int nb_db_transaction_save(const struct nb_transaction *transaction,
 	 * values too, as this covers the case where defaults may change.
 	 */
 	if (lyd_print_mem(&config_str, transaction->config->dnode, LYD_XML,
-			  LYD_PRINT_WITHSIBLINGS | LYD_PRINT_WD_ALL)
+			  LYD_PRINT_SIBLINGS | LYD_PRINT_WD_ALL)
 	    != 0)
 		goto exit;
 
diff --git a/lib/northbound_grpc.cpp b/lib/northbound_grpc.cpp
index 6c33351cef..b3e14828f3 100644
--- a/lib/northbound_grpc.cpp
+++ b/lib/northbound_grpc.cpp
@@ -370,7 +370,7 @@ static LY_ERR data_tree_from_dnode(frr::DataTree *dt,
 	char *strp;
 	int options = 0;
 
-	SET_FLAG(options, LYD_PRINT_WITHSIBLINGS);
+	SET_FLAG(options, LYD_PRINT_SIBLINGS);
 	if (with_defaults)
 		SET_FLAG(options, LYD_PRINT_WD_ALL);
 	else
diff --git a/lib/yang.c b/lib/yang.c
index 4dd8654217..44563d4922 100644
--- a/lib/yang.c
+++ b/lib/yang.c
@@ -10,11 +10,23 @@
 #include "lib_errors.h"
 #include "yang.h"
 #include "yang_translator.h"
+#include <libyang/version.h>
 #include "northbound.h"
 
 DEFINE_MTYPE_STATIC(LIB, YANG_MODULE, "YANG module");
 DEFINE_MTYPE_STATIC(LIB, YANG_DATA, "YANG data structure");
 
+/* Safe to remove after libyang 2.2.8 */
+#if (LY_VERSION_MAJOR < 3)
+#define yang_lyd_find_xpath3(ctx_node, tree, xpath, format, prefix_data, vars, \
+			     set)                                              \
+	lyd_find_xpath3(ctx_node, tree, xpath, vars, set)
+#else
+#define yang_lyd_find_xpath3(ctx_node, tree, xpath, format, prefix_data, vars, \
+			     set)                                              \
+	lyd_find_xpath3(ctx_node, tree, xpath, LY_VALUE_JSON, NULL, vars, set)
+#endif
+
 /* libyang container. */
 struct ly_ctx *ly_native_ctx;
 
@@ -329,9 +341,7 @@ const char *yang_snode_get_default(const struct lysc_node *snode)
 	switch (snode->nodetype) {
 	case LYS_LEAF:
 		sleaf = (const struct lysc_node_leaf *)snode;
-		return sleaf->dflt ? lyd_value_get_canonical(sleaf->module->ctx,
-							     sleaf->dflt)
-				   : NULL;
+		return sleaf->dflt.str;
 	case LYS_LEAFLIST:
 		/* TODO: check leaf-list default values */
 		return NULL;
@@ -657,7 +667,12 @@ struct yang_data *yang_data_list_find(const struct list *list,
 }
 
 /* Make libyang log its errors using FRR logging infrastructure. */
-static void ly_log_cb(LY_LOG_LEVEL level, const char *msg, const char *path)
+static void ly_zlog_cb(LY_LOG_LEVEL level, const char *msg, const char *data_path
+#if !(LY_VERSION_MAJOR < 3)
+		       ,
+		       const char *schema_path, uint64_t line
+#endif
+)
 {
 	int priority = LOG_ERR;
 
@@ -674,8 +689,14 @@ static void ly_log_cb(LY_LOG_LEVEL level, const char *msg, const char *path)
 		break;
 	}
 
-	if (path)
-		zlog(priority, "libyang: %s (%s)", msg, path);
+	if (data_path)
+		zlog(priority, "libyang: %s (%s)", msg, data_path);
+#if !(LY_VERSION_MAJOR < 3)
+	else if (schema_path)
+		zlog(priority, "libyang %s (%s)\n", msg, schema_path);
+	else if (line)
+		zlog(priority, "libyang %s (line %" PRIu64 ")\n", msg, line);
+#endif
 	else
 		zlog(priority, "libyang: %s", msg);
 }
@@ -752,7 +773,7 @@ struct ly_ctx *yang_ctx_new_setup(bool embedded_modules, bool explicit_compile)
 void yang_init(bool embedded_modules, bool defer_compile)
 {
 	/* Initialize libyang global parameters that affect all containers. */
-	ly_set_log_clb(ly_log_cb, 1);
+	ly_set_log_clb(ly_zlog_cb);
 	ly_log_options(LY_LOLOG | LY_LOSTORE);
 
 	/* Initialize libyang container for native models. */
