diff --git a/meson.build b/meson.build
index 8cddf77..e97f862 100644
--- a/meson.build
+++ b/meson.build
@@ -188,7 +188,7 @@ endif
 config_h.set('WITH_UDEV', enable_udev)

 # udev base directory (required to install rules even when udev support is disabled)
-udev_udevdir = get_option('udevdir')
+udev_udevdir = '/lib/udev'
 if udev_udevdir == ''
   assert(enable_udev, 'udevdir must be explicitly given if udev support is disabled')
   udev_udevdir = dependency('udev').get_pkgconfig_variable('udevdir')
