diff --git a/src/mm-bearer-qmi.c b/src/mm-bearer-qmi.c
index 54f2e93..3c4df7b 100644
--- a/src/mm-bearer-qmi.c
+++ b/src/mm-bearer-qmi.c
@@ -2494,7 +2494,11 @@ load_settings_from_bearer (MMBearerQmi         *self,
         }
 
         /* The link prefix hint given must be modem-specific */
-        ctx->link_prefix_hint = g_strdup_printf ("qmapmux%u.", mm_base_modem_get_dbus_id (MM_BASE_MODEM (modem)));
+        const gchar *name = mm_port_get_device (MM_PORT (ctx->data));
+        if (name)
+                ctx->link_prefix_hint = g_strdup_printf ("%s.", name);
+        else
+                ctx->link_prefix_hint = g_strdup_printf ("qmapmux%u.", mm_base_modem_get_dbus_id (MM_BASE_MODEM (modem)));
     }
 
     /* If profile id is given, we'll launch the connection specifying the profile id in use
