From c15bb6208cdd0e0629a28c10cb448a11aa7a1624 Mon Sep 17 00:00:00 2001
From: Joachim Wiberg <troglobit@gmail.com>
Date: Mon, 15 Jul 2024 14:24:08 +0200
Subject: [PATCH 08/17] Fix gcc warning in vsyslogp_r()
Organization: Addiva Elektronik

Signed-off-by: Joachim Wiberg <troglobit@gmail.com>
---
 src/syslog.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/syslog.c b/src/syslog.c
index 1510258..fbbdfdf 100644
--- a/src/syslog.c
+++ b/src/syslog.c
@@ -267,7 +267,7 @@ vsyslogp_r(int pri, struct syslog_data *data, const char *msgid,
 	/* Default log format is RFC5424, continues below BSD format */
 	if (data->log_stat & LOG_RFC3164) {
 		const char *tag = data->log_tag;
-		char tmp[33];
+		char tmp[256];
 
 		if (!(data->log_stat & LOG_NLOG)) {
 			prlen = snprintf(p, tbuf_left, "<%d>", pri);
@@ -303,12 +303,12 @@ vsyslogp_r(int pri, struct syslog_data *data, const char *msgid,
 
 		/*
 		 * When sending remote we MUST follow RFC3164 sec 4.1.3,
-		 * otherwise we "cheat" and allow max lenght hostname,
+		 * otherwise we "cheat" and allow max length hostname,
 		 * for either log file or local syslogd -- it is up to
 		 * the local syslogd then to fulfill RFC req. on output
 		 */
 		if (data->log_host) {
-			strlcpy(tmp, data->log_tag, sizeof(tbuf));
+			strlcpy(tmp, data->log_tag, sizeof(tmp));
 			tag = tmp;
 		}
 
-- 
2.43.0

